internal/core/ui/src/main/kotlin/com/recco/internal/core/ui/components/AppAsyncImage.kt

```diff
@@ -66,16 +68,17 @@ fun AppAsyncImage(
             onStateChange(state)
 
             Image(
+                modifier = modifier
+                    .let {
+                        if (aspectRatio != null) {
+                            it.aspectRatio(aspectRatio)
+                        } else {
+                            it
+                        }
+                    },
                 painter = painter,
                 contentScale = contentScale,
-                contentDescription = null,
-                modifier = modifier.let {
-                    if (aspectRatio != null) {
-                        it.aspectRatio(aspectRatio)
-                    } else {
-                        it
-                    }
-                }
+                contentDescription = null
```

Comment:
I initially filled that `contentDescription` with the title of the card, detail page,... but I remove it eventually because I though it was not needed to repeat the same content description when using TalkBack.

Looking at the doc it says:
<img width="568" alt="Screenshot 2023-08-01 at 09 18 56" src="https://github.com/sf-recco/android-sdk/assets/2734557/05ce19b7-fcf2-4c20-9091-ec855e1c8f6e">

so I think for our case is not needed because the image always goes with a text...

The ways of testing or analyzing it are the ones described here  https://developer.android.com/guide/topics/ui/accessibility/testing. I used the two first ones (TalkBack and Accessibility Scanner)

Also in some cases I use `.semantics(mergeDescendants = true) {}` to let some widgets to use the same text in one child within the parent, for example a card containing image and text.

<img width="636" alt="Screenshot 2023-08-01 at 09 33 32" src="https://github.com/sf-recco/android-sdk/assets/2734557/58b07f67-1409-4fd9-bdb8-d464b9e3775f">

Reply:
I see, thanks for the clarification!